x-pack/plugins/security_solution/public/dashboards/components/edit_dashboard_button.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { Query, Filter } from '@kbn/es-query';
+import { useDashboardAppLink } from '../hooks/use_dashboard_app_link';
+import { EDIT_DASHBOARD_BUTTON_TITLE } from '../pages/details/translations';
+import { useKibana } from '../../common/lib/kibana/kibana_react';
+import { LinkButton } from '../../common/components/links';
+
+export interface EditDashboardButtonComponentProps {
+  dashboardExists: boolean;
+  filters?: Filter[];
+  query?: Query;
+  savedObjectId: string | undefined;
+  showWriteControls: boolean;
+  timeRange: {
+    from: string;
+    to: string;
+    fromStr?: string | undefined;
+    toStr?: string | undefined;
+  };
+}
+
+const EditDashboardButtonComponent: React.FC<EditDashboardButtonComponentProps> = ({
+  dashboardExists,
+  filters,
+  query,
+  savedObjectId,
+  showWriteControls,
+  timeRange,
+}) => {
+  const {
+    services: { uiSettings },
+  } = useKibana();
+
+  const editDashboardUrl = useDashboardAppLink({
+    query,
+    filters,
+    timeRange,
+    uiSettings,
+    savedObjectId,
+  });
+
+  return showWriteControls && dashboardExists ? (
+    <LinkButton
+      color="primary"
+      fill
+      iconType="pencil"
+      href={editDashboardUrl}
```

Comment:
We are doing a hard redirect here. We need to set the `onClick` to the link and navigate using history. We can use `useLinkProps({ appId: DASHBOARD_APP_ID, path: '...' })` inside the `useDashboardAppLink` hook, and return both, the `href` and the `onClick`, and add both props to this link.

Reply:
Yup, it's now navigating to the page by navigateTo, no hard refresh.